engine.py

```diff
@@ -348,6 +348,39 @@ def post_app_init(self):
             # hook things up so that the menu is created every time it is clicked
             self._menu_handle.postMenuCommand(self._menu_generator.create_menu)
 
+        # Build a dictionary mapping app instance names to lists of commands they registered with the engine.
```

Comment:
given that it's a fair bit of code, i think we should wrap it in its own method rather than having it directly in post_app_init.